src/gui/plugins/apply_force_torque/ApplyForceTorque.cc

```diff
@@ -398,5 +566,343 @@ void ApplyForceTorquePrivate::PublishWrench(bool _applyForce, bool _applyTorque)
   this->pub.Publish(msg);
 }
 
+/////////////////////////////////////////////////
+void ApplyForceTorquePrivate::OnRender()
+{
+  if (!this->scene)
+  {
+    // Get scene and user camera
+    this->scene = rendering::sceneFromFirstRenderEngine();
+    if (!this->scene)
+    {
+      return;
+    }
+
+    for (unsigned int i = 0; i < this->scene->NodeCount(); ++i)
+    {
+      auto cam = std::dynamic_pointer_cast<rendering::Camera>(
+        this->scene->NodeByIndex(i));
+      if (cam && cam->HasUserData("user-camera") &&
+          std::get<bool>(cam->UserData("user-camera")))
+      {
+        this->camera = cam;
+        gzdbg << "ApplyForceTorque plugin is using camera ["
+              << this->camera->Name() << "]" << std::endl;
+        break;
+      }
+    }
+
+    this->ray = this->scene->CreateRayQuery();
+
+    // Make material into overlay
+    auto mat = this->scene->Material("Default/TransRed")->Clone();
+    mat->SetDepthCheckEnabled(false);
+    mat->SetDepthWriteEnabled(false);
+
+    // Create visuals
+    this->wrenchVis.Init(this->scene);
+    this->forceVisual = this->wrenchVis.CreateForceVisual(mat);
+    this->torqueVisual = this->wrenchVis.CreateTorqueVisual(mat);
+    this->gizmoVisual = this->scene->CreateGizmoVisual();
+    this->scene->RootVisual()->AddChild(this->gizmoVisual);
+  }
+
+  this->HandleMouseEvents();
+
+  this->UpdateVisuals();
+
+  if (this->sendBlockOrbit)
+  {
+    // Events with false should only be sent once
+    if (!this->blockOrbit)
+      this->sendBlockOrbit = false;
+
+    gz::gui::events::BlockOrbit blockOrbitEvent(this->blockOrbit);
+    gz::gui::App()->sendEvent(
+        gz::gui::App()->findChild<gz::gui::MainWindow *>(),
+        &blockOrbitEvent);
+  }
```

Comment:
Why does this plugin need to send `BlockOribit`? Does the user interact with it in the 3D scene?

Edit: Ah, after reading the description, I see that there's supposed to be some interaction. However, this is not working for me. Anytime I interact with the 3D scene, the Orbit view is taking over.